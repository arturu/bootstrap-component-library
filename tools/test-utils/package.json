{
  "name": "@openeuropa/bcl-test-utils",
  "version": "0.1.3",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
