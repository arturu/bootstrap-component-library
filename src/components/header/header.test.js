import { renderTwigFileAsNode } from "@oe/test-utils";

import demoData from "@oe/data-header/data";
import demoDataEcl from "@oe/data-header/data-ecl";

const template = "@oe/header/header.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Header", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});

describe("OE - Header ECL", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoDataEcl)).resolves.toMatchSnapshot();
  });
});
