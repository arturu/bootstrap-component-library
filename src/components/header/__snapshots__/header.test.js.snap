// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Header ECL renders correctly 1`] = `
<header>
  <!-- header -->
  <div
    class="container"
  >
    <div
      class="oe-header"
    >
      <a
        class="navbar-brand"
        href="#"
      >
        <img
          height="60"
          src="https://ec.europa.eu/info/sites/info/themes/europa/images/svg/logo/logo--en.svg"
          width="240"
        />
      </a>
      <ul
        class="nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-bs-target="#loginModal"
            data-bs-toggle="modal"
            href="Log"
            in=""
            role="button"
          >
            
            Log in
          
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-bs-target="#languageModal"
            data-bs-toggle="modal"
            href="English"
            role="button"
          >
            
            English
          
          </a>
        </li>
      </ul>
    </div>
  </div>
  <!-- navbar -->
  <nav
    class="oe-navbar navbar navbar-dark navbar-expand-lg"
  >
    <div
      class="container"
    >
      <button
        aria-controls="navbarNavDropdown"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarNavDropdown"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarNavDropdown"
      >
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="#"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Item 01
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Item 02
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdownMenuLink"
              role="button"
            >
              
            Dropdown link
          
            </a>
            <ul
              aria-labelledby="navbarDropdownMenuLink"
              class="dropdown-menu"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Another action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Something else here
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;

exports[`OE - Header renders correctly 1`] = `
<header>
  <!-- header -->
  <div
    class="container"
  >
    <div
      class="oe-header"
    >
      <a
        class="navbar-brand"
        href="#"
      >
        <img
          height="60"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/3f/Placeholder_view_vector.svg/638px-Placeholder_view_vector.svg.png"
          width="100"
        />
      </a>
      <ul
        class="nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-bs-target="#loginModal"
            data-bs-toggle="modal"
            href="Log"
            in=""
            role="button"
          >
            
            Log in
          
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-bs-target="#languageModal"
            data-bs-toggle="modal"
            href="English"
            role="button"
          >
            
            English
          
          </a>
        </li>
      </ul>
    </div>
  </div>
  <!-- navbar -->
  <nav
    class="oe-navbar navbar navbar-light navbar-expand-lg"
  >
    <div
      class="container"
    >
      <button
        aria-controls="navbarNavDropdown"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarNavDropdown"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarNavDropdown"
      >
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="#"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Item 01
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Item 02
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdownMenuLink"
              role="button"
            >
              
            Dropdown link
          
            </a>
            <ul
              aria-labelledby="navbarDropdownMenuLink"
              class="dropdown-menu"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Another action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Something else here
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;
